### ISSUE TITLE
Memory allocation fails on Windows 7 SP1

### ISSUE THREAD

**Issue by szarnyasg** [duckdb/duckdb#3079]
#### What happens?

DuckDB fails when trying to allocate memory on Windows 7.

#### To Reproduce

Grab the DuckDB Windows executable. I tried both the 32- and the 64-bit CLI variants, and also tried the 64-bit variant from Python 3.8.

```console
$ duckdb.exe
v0.3.2 5aebf7dac
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D select 42;
Error: unable to open database ":memory:": IO Error: Could not fetch available system memory!
```

The same error occurs when trying to use a database file except `:memory:` is replaced by the database file's name in the error message.

I'm sure running on the Windows 7 OS is now quite rare among DuckDB users. So I'm mainly raising this issue so others can find it if they stumble upon it.

#### Environment (please complete the following information):
 - OS: Windows 7 SP1 running in VirtualBox on a Fedora 35 host
 - DuckDB Version: 0.3.2
 - DuckDB Client: CLI

#### Before Submitting

- [ ] **Have you tried this on the latest `master` branch?** no

**Comment by szarnyasg** [comment]
I have to add that I'm running Windows 7 in VirtualBox on a Fedora 35 host. I edited the original post to add this information.

I tried building a fresh DuckDB binary with Cygwin, CMake and GCC 11. However, the build failed:

```console
Consolidate compiler generated dependencies of target duckdb_common
[ 33%] Building CXX object src/common/CMakeFiles/duckdb_common.dir/ub_duckdb_common.cpp.o
In file included from /cygdrive/c/Users/Gabor/Downloads/duckdb/build/src/common/ub_duckdb_common.cpp:16:
/cygdrive/c/Users/Gabor/Downloads/duckdb/src/common/local_file_system.cpp: In member function ‘virtual void duckdb::LocalFileSystem::Read(duckdb::FileHandle&, void*, int64_t, duckdb::idx_t)’:
/cygdrive/c/Users/Gabor/Downloads/duckdb/src/common/local_file_system.cpp:203:30: error: ‘pread’ was not declared in this scope; did you mean ‘read’?
  203 |         int64_t bytes_read = pread(fd, buffer, nr_bytes, location);
      |                              ^~~~~
      |                              read
/cygdrive/c/Users/Gabor/Downloads/duckdb/src/common/local_file_system.cpp: In member function ‘virtual void duckdb::LocalFileSystem::Write(duckdb::FileHandle&, void*, int64_t, duckdb::idx_t)’:
/cygdrive/c/Users/Gabor/Downloads/duckdb/src/common/local_file_system.cpp:224:33: error: ‘pwrite’ was not declared in this scope; did you mean ‘write’?
  224 |         int64_t bytes_written = pwrite(fd, buffer, nr_bytes, location);
      |                                 ^~~~~~
      |                                 write
/cygdrive/c/Users/Gabor/Downloads/duckdb/src/common/local_file_system.cpp: In member function ‘virtual void duckdb::LocalFileSystem::Truncate(duckdb::FileHandle&, int64_t)’:
/cygdrive/c/Users/Gabor/Downloads/duckdb/src/common/local_file_system.cpp:268:13: error: ‘ftruncate’ was not declared in this scope; did you mean ‘Truncate’?
  268 |         if (ftruncate(fd, new_size) != 0) {
      |             ^~~~~~~~~
      |             Truncate
make[2]: *** [src/common/CMakeFiles/duckdb_common.dir/build.make:76: src/common/CMakeFiles/duckdb_common.dir/ub_duckdb_common.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:4181: src/common/CMakeFiles/duckdb_common.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by Mytherin** [comment]
> Error: unable to open database ":memory:": IO Error: Could not fetch available system memory!

This error message occurs when the system call `GetPhysicallyInstalledSystemMemory` returns false. I imagine this is likely caused by the VM setup, particularly there is this section in [the docs](https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getphysicallyinstalledsystemmemory):

```
Return value
If the function succeeds, it returns TRUE and sets the TotalMemoryInKilobytes parameter to a nonzero value.
If the function fails, it returns FALSE and does not modify the TotalMemoryInKilobytes parameter. To get extended error information, use the [GetLastError](https://docs.microsoft.com/en-us/windows/desktop/api/errhandlingapi/nf-errhandlingapi-getlasterror) function. Common errors are listed in the following table.

ERROR_INVALID_DATA | The System Management BIOS (SMBIOS) data is malformed.
```

Perhaps the Windows install is malformed, or there is another problem with using this function to get the available memory in a VM? I imagine the CI jobs we run also run on VMs, however, and it seems to work there.

> I tried building a fresh DuckDB binary with Cygwin, CMake and GCC 11. However, the build failed:

This seems to be caused by `_WIN32` not being defined. Perhaps we are relying on a define specific to MSVC to detect whether we are running on Windows there. I will have a look.

**Comment by Shawn201107** [comment]
@Mytherin Did you find any solution? I also had a similar issue.

- OS: Windows Server 2012 R2
- DuckDB version: 0.3.1
- DuckDB client: JDBC

```
java.sql.SQLException: IO Error: Could not fetch available system memory!
at org.duckdb.DuckDBNative.duckdb_jdbc_startup(Native Method) ~[test-service.jar:?]
at org.duckdb.DuckDBDatabase.<init>(DuckDBDatabase.java:22) ~[test-service.jar:?]
at org.duckdb.DuckDBDriver.connect(DuckDBDriver.java:35) ~[test-service.jar:?]
at java.sql.DriverManager.getConnection(DriverManager.java:664) ~[?:1.8.0_202]
at java.sql.DriverManager.getConnection(DriverManager.java:270) ~[?:1.8.0_202]
```

**Comment by Mytherin** [comment]
`GetAvailableMemory()` is only called when `max_memory` is not set in the config. If you specify a `max_memory` on start-up then the function should never be called and you won't get this error. I am not sure if this configuration flag is exposed in the JDBC client, however.

**Comment by Shawn201107** [comment]
The configuration does not seem to be exposed in the JDBC client.

**Comment by ededovic** [comment]
Do you have a suggestion how to get around this issue? What should I ask to be set up on the VM, so that that function gets memory installed? I'm getting similar error on windows server when trying to import:

`release='2012ServerR2', version='6.3.9600'`

`In [4]: import duckdb
ImportError                               Traceback (most recent call last)
<ipython-input-4-68831cbe5a63> in <module>()
----> 1 import duckdb
...\envs\py36\lib\site-packages\duckdb\__init__.py in <module>()
      8     sys.setdlopenflags(os.RTLD_GLOBAL | os.RTLD_NOW)
      9
---> 10 from _duckdb_extension import *
     11 import _duckdb_extension
     12 __all__ = [name for name in dir(_duckdb_extension) if name[0] != '_']`
`ImportError: IO Error: Could not fetch available system memory!`

**Comment by ilcaa72** [comment]
i received this memory error message using **DBeaver**.  I am on Windows 10.
![dbeaver_Jdbc driver](https://user-images.githubusercontent.com/63480773/175033749-9b9de109-2d4f-4d38-86cf-e112afc3fdba.jpg)

**Comment by ededovic** [comment]
Would it be possible to change the code such that if the physical memory can not be determined to assume say 1GB and print out a warning. At the moment, I simply can not use duckdb on those machine. If import worked,  I can use pragma to change the memory size.

**Comment by Mytherin** [comment]
Yes, that would be possible and is a good idea.

**Comment by Mytherin** [comment]
Fixed by #3971, we no longer throw an error when we cannot get the system memory and instead default to no memory limit.

**Comment by abubelinha** [comment]
@szarnyasg can you confirm whether DuckDB should run in Windows7 or not?
Are there any specific system requirements?

I am trying to run this version on Windows 7 Home Premium SP1 64bit:
https://github.com/duckdb/duckdb/releases/download/v1.0.0/duckdb_cli-windows-amd64.zip

I don't get your memory error, but a message box says VCRUNTIME140_1.dll is missing 
(although Microsoft Visual C++ 2015 Redistributables is installed -both x64 and x86-)

Thanks
@abubelinha
